data-prepper-plugins/aws-lambda/src/test/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessorTest.java

```diff
@@ -1428,6 +1431,25 @@ private Record<Event> createLargeRecord(final int sizeInBytes) {
         return new Record<>(event);
     }
 
+    @Test
+    void testNoClassCastExceptionWithAggregateEventHandle() {
+        // Simulate event from aggregate processor with AggregateEventHandle
+        Event mockEvent = mock(Event.class);
+        Record<Event> mockRecord = mock(Record.class);
+        AggregateEventHandle aggregateHandle = new AggregateEventHandle(Instant.now());
+        
+        when(mockEvent.getEventHandle()).thenReturn(aggregateHandle);
+        when(mockRecord.getData()).thenReturn(mockEvent);
+
+        AggregateResponseEventHandlingStrategy strategy = new AggregateResponseEventHandlingStrategy();
+        List<Event> parsedEvents = Collections.emptyList();
+        List<Record<Event>> originalRecords = Collections.singletonList(mockRecord);
+
+        assertDoesNotThrow(() -> {
```

Comment:
Rather than making this test oriented around this not throwing, we should be testing the desired behavior when it is an `AggregateEventHandle`.

Use Mockito to verify this interaction:

```
aggregateHandle.addEventHandle(responseEvent.getEventHandle());
```